src/main/java/com/nulabinc/zxcvbn/matchers/Dictionary.java

```diff
@@ -1,60 +1,42 @@
 package com.nulabinc.zxcvbn.matchers;
 
-import java.io.BufferedReader;
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.InputStreamReader;
-import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
 public class Dictionary {
 
-    private static final String RESOURCES_PACKAGE_PATH = "/com/nulabinc/zxcvbn/matchers/dictionarys/";
+    private final String name;
 
-    private static final ResourceLoader RESOURCE_LOADER = new ResourceLoader();
+    private final List<String> frequencies;
 
-    private static final String EXT = ".txt";
+    private final Map<String, Integer> rankedDictionary;
 
-    private static final String UTF_8 = "UTF-8";
+    public Dictionary(String name, List<String> frequencies) {
+        this.name = name;
+        this.frequencies = frequencies;
+        this.rankedDictionary = toRankedDictionary(frequencies);
+    }
 
+    private Map<String, Integer> toRankedDictionary(final List<String> frequencies) {
+        Map<String, Integer> result = new HashMap<>();
+        int i = 1; // rank starts at 1, not 0
+        for (String word : frequencies) {
+            result.put(word, i);
+            i++;
+        }
+        return result;
+    }
 
-    private static String buildResourcePath(String filename) {
-        return RESOURCES_PACKAGE_PATH + filename + EXT;
+    public String getName() {
+        return name;
     }
 
-    private static final String[] DICTIONARY_PARAMS = {
-            "us_tv_and_film",
-            "english_wikipedia",
-            "passwords",
-            "surnames",
-            "male_names",
-            "female_names"
-    };
-
-    public static final Map<String, String[]> FREQUENCY_LISTS;
-    static {
-        FREQUENCY_LISTS = read();
+    public List<String> getFrequencies() {
+        return frequencies;
     }
 
-    private static Map<String, String[]> read() {
-        Map<String, String[]> freqLists = new HashMap<>();
-        for (String filename:  DICTIONARY_PARAMS) {
-            List<String> words = new ArrayList<>();
-            try(InputStream is = RESOURCE_LOADER.getInputStream(buildResourcePath(filename));
-                // Reasons for not using StandardCharsets
-                // refs: https://github.com/nulab/zxcvbn4j/issues/62
-                BufferedReader br = new BufferedReader(new InputStreamReader(is, UTF_8))) {
-                String line;
-                while ((line = br.readLine()) != null) {
-                    words.add(line);
-                }
-            } catch (IOException e) {
-                throw new RuntimeException("Error while reading " + filename);
-            }
-            freqLists.put(filename, words.toArray(new String[]{}));
-        }
-        return freqLists;
+    public Map<String, Integer> getRankedDictionary() {
+        return rankedDictionary;
     }
 }
```

Comment:
POSIX rules require a newline at the end of the file. Please add a newline.
<img width="165" alt="image" src="https://user-images.githubusercontent.com/9883589/162594418-8f1c6182-ffb6-43f5-b097-9bedf36e9170.png">